packages/injective-test-tube/src/module/authz.rs

```diff
@@ -0,0 +1,211 @@
+                msg: "/injective.exchange.v1beta1.MsgCreateSpotLimitOrder".to_string(),
+            },
+            &mut buf_2,
+        )
+        .unwrap();
+
+        authz
+            .grant(
+                MsgGrant {
+                    granter: signer.address(),
+                    grantee: receiver.address(),
+                    grant: Some(Grant {
+                        authorization: Some(Any {
+                            type_url: "/cosmos.authz.v1beta1.GenericAuthorization".to_string(),
+                            value: buf_2.clone(),
+                        }),
+                        expiration: None,
+                    }),
+                },
+                &signer,
+            )
+            .unwrap();
+
+        let response = authz
+            .query_grantee_grants(&QueryGranteeGrantsRequest {
+                grantee: receiver.address(),
+                pagination: None,
+            })
+            .unwrap();
+        assert_eq!(
+            response.grants,
+            vec![
+                GrantAuthorization {
+                    granter: signer.address(),
+                    grantee: receiver.address(),
+                    authorization: Some(Any {
+                        type_url: "/cosmos.bank.v1beta1.SendAuthorization".to_string(),
+                        value: buf.clone(),
+                    }),
+                    expiration: None,
+                },
+                GrantAuthorization {
+                    granter: signer.address(),
+                    grantee: receiver.address(),
+                    authorization: Some(Any {
+                        type_url: "/cosmos.authz.v1beta1.GenericAuthorization".to_string(),
+                        value: buf_2.clone(),
+                    }),
+                    expiration: None,
+                }
+            ]
+        );
+
+        let response = authz
+            .query_granter_grants(&QueryGranterGrantsRequest {
+                granter: signer.address(),
+                pagination: None,
+            })
+            .unwrap();
+        assert_eq!(
+            response.grants,
+            vec![
+                GrantAuthorization {
+                    granter: signer.address(),
+                    grantee: receiver.address(),
+                    authorization: Some(Any {
+                        type_url: "/cosmos.bank.v1beta1.SendAuthorization".to_string(),
+                        value: buf,
+                    }),
+                    expiration: None,
+                },
+                GrantAuthorization {
+                    granter: signer.address(),
+                    grantee: receiver.address(),
+                    authorization: Some(Any {
+                        type_url: "/cosmos.authz.v1beta1.GenericAuthorization".to_string(),
+                        value: buf_2,
+                    }),
+                    expiration: None,
+                }
+            ]
+        );
+    }
```

Comment:
maybe add some comments that explain what each grant is attempting to do and what is the expected result? Or split them into various and aptly named tests?

what about some negative cases like:
* type_url being empty / random string
* value being just random strings?